saleor/core/utils/__init__.py

```diff
@@ -70,14 +72,24 @@ def get_currency_for_country(country):
     return settings.DEFAULT_CURRENCY
 
 
-def get_paginator_items(items, paginate_by, page):
+def get_paginator_items(items, paginate_by, page_number):
+    if not page_number:
+        page_number = 1
     paginator = Paginator(items, paginate_by)
     try:
-        items = paginator.page(page)
-    except PageNotAnInteger:
-        items = paginator.page(1)
-    except EmptyPage:
-        items = paginator.page(paginator.num_pages)
+        page_number = int(page_number)
+    except ValueError:
+        raise Http404(pgettext_lazy(
+            'Message related to an pagination',
+            'Page can not be converted to an int.'))
```

Comment:
I don't think `Http404` are visible to users? There's not reason to translate system exceptions.